doc/src/manual/interfaces.md

```diff
@@ -136,6 +136,25 @@ define an informal interface that enable many fancier behaviors. In some cases,
 to additionally specialize those extra behaviors when they know a more efficient algorithm can
 be used in their specific case.
 
+It is also often useful to allow iteration over a collection in *reverse order*
+by iterating over `Reversed(iterator)`.  [`Reversed`](@ref) is just a thin wrapper
+around our collection; to actually support reverse-order iteration, an iterator
+type `T` needs to implement `start`, `next`, and `done` methods for `Reversed{T}`.
+(Given `r::Reversed{T}`, the underling iterator of type `T` is `Reversed(r)`.)
```

Comment:
Should we just document `r.itr` instead of suggesting `Reversed(r)` or `Iterators.reverse(r)` or whatever we decide to call it?